api/src/pcapi/routes/pro/offers.py

```diff
@@ -95,22 +95,27 @@ def get_stocks(offer_id: int, query: offers_serialize.StocksQueryModel) -> offer
             status_code=404,
         )
     rest.check_user_has_access_to_offerer(current_user, offer.venue.managingOffererId)
-    stocks = offers_repository.get_filtered_stocks(
-        offer_id=offer_id,
-        date=query.date,
-        time=query.time,
-        price_category_id=query.price_category_id,
-        order_by=query.order_by,
-        order_by_desc=query.order_by_desc,
-    )
-    stocks_count = stocks.count()
-    stocks = offers_repository.get_paginated_stocks(
-        stocks_query=stocks,
-        page=query.page,
-        stocks_limit_per_page=query.stocks_limit_per_page,
-    )
-    stock_list = [offers_serialize.GetOfferStockResponseModel.from_orm(stock) for stock in stocks.all()]
-    return offers_serialize.GetStocksResponseModel(stocks=stock_list, stock_count=stocks_count)
+    has_stocks = offers_repository.offer_has_stocks(offer_id=offer_id)
+    if has_stocks:
+        stocks = offers_repository.get_filtered_stocks(
+            offer_id=offer_id,
+            date=query.date,
+            time=query.time,
+            price_category_id=query.price_category_id,
+            order_by=query.order_by,
+            order_by_desc=query.order_by_desc,
+        )
+        stocks_count = stocks.count()
+        stocks = offers_repository.get_paginated_stocks(
+            stocks_query=stocks,
+            page=query.page,
+            stocks_limit_per_page=query.stocks_limit_per_page,
+        )
+        stock_list = [offers_serialize.GetOfferStockResponseModel.from_orm(stock) for stock in stocks.all()]
+    else:
+        stock_list = []
+        stocks_count = 0
+    return offers_serialize.GetStocksResponseModel(stocks=stock_list, stockCount=stocks_count, hasStocks=has_stocks)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Ça me parait très bizarre de s'embèter à re-calculer `offer_has_stocks` 
ici tu as déjà stock_count

```suggestion
    return offers_serialize.GetStocksResponseModel(stocks=stock_list, stockCount=stocks_count, hasStocks=stocks_count>0)